### ISSUE TITLE
Xephyr big window

### ISSUE THREAD

**Issue by mbnoimi** [clearkimura/Customizer#48]
How can I change Xephyr window size?

I set the resolution of Xephyr to 800x600 (default value) but when I run it; It shows very big window so I can't modify anything through it!

**Comment by ghost** [comment]
@mbnoimi 
On our side, we already tested Customizer with various ISO,
 and we don't face problem at all with Xephyr so far.

This is why we must remind users again and again:  details, please!

Make it a habit to provide the details (also mentioned in our user guide).
Otherwise, we don't know how you face this issue after all?

**Comment by mbnoimi** [comment]
@clearkimura 

> This is why we must remind users again and again: details, please!

My screen resolution is 1440 x 900 (16:10) while Xephyr appears in resolution more than 1440 x 900 although I configured it to appear in 800 x 600 as shown below:
![screenshot from 2014-01-28 17 35 01](https://f.cloud.github.com/assets/284547/2020400/f1b46852-8831-11e3-9f23-f6c7ce117ef1.png)

**Comment by mbnoimi** [comment]
the log of Xephyr

```
[*] Executing xnest at: CET 05:35:44 /28-01-2014
[*] Checking filesystem
[*] Checking lock
[!] Multiple X-sessions available, choose which one to execute:

cinnamon2d
cinnamon

[?] What shall it be? cinnamon
[*] Starting Xephyr
[*] Allwoing local access to X-server
non-network local connections being added to access control list
[*] Mounting: /dev
[*] Mounting: /proc
[*] Mounting: /sys
[*] Mounting: /var/lib/dbus
[*] Mounting: /run/dbus
[*] Doing some preparations
[*] Setting up locales
Initializing built-in extension Generic Event Extension
Initializing built-in extension SHAPE
Initializing built-in extension MIT-SHM
Initializing built-in extension XInputExtension
Initializing built-in extension XTEST
Initializing built-in extension BIG-REQUESTS
Initializing built-in extension SYNC
Initializing built-in extension XKEYBOARD
Initializing built-in extension XC-MISC
Initializing built-in extension SECURITY
Initializing built-in extension XINERAMA
Initializing built-in extension XFIXES
Initializing built-in extension RENDER
Initializing built-in extension RANDR
Initializing built-in extension COMPOSITE
Initializing built-in extension DAMAGE
Initializing built-in extension MIT-SCREEN-SAVER
Initializing built-in extension DOUBLE-BUFFER
Initializing built-in extension RECORD
Initializing built-in extension DPMS
Initializing built-in extension X-Resource
Initializing built-in extension XVideo
Initializing built-in extension XVideo-MotionCompensation
Initializing built-in extension SELinux
Initializing built-in extension GLX
[dix] Could not init font path element /usr/share/fonts/X11/cyrillic, removing from list!
[dix] Could not init font path element /usr/share/fonts/X11/100dpi/:unscaled, removing from list!
[dix] Could not init font path element /usr/share/fonts/X11/75dpi/:unscaled, removing from list!
[dix] Could not init font path element /usr/share/fonts/X11/100dpi, removing from list!
[dix] Could not init font path element /usr/share/fonts/X11/75dpi, removing from list!
[*] Blocking files
[*] Updating packages database
[*] Making sure everything is configured
Reading package lists...
Building dependency tree...
Reading state information...
0 upgraded, 0 newly installed, 0 to remove and 189 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
dbus is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 189 not upgraded.
[*] Unblocking files
[*] Cleaning up
[*] Doing some preparations
[*] Setting up locales
[*] Blocking files
[*] Updating packages database
[*] Making sure everything is configured
Reading package lists...
Building dependency tree...
Reading state information...
0 upgraded, 0 newly installed, 0 to remove and 189 not upgraded.
[*] Unblocking files
[*] Cleaning up
[*] Doing some preparations
[*] Setting up locales
[*] Blocking files
[*] Updating packages database
[*] Making sure everything is configured
Reading package lists...
Building dependency tree...
Reading state information...
0 upgraded, 0 newly installed, 0 to remove and 189 not upgraded.
GNOME_KEYRING_CONTROL=/root/.cache/keyring-aZPaEg
GNOME_KEYRING_PID=4822
GNOME_KEYRING_CONTROL=/root/.cache/keyring-aZPaEg
GNOME_KEYRING_CONTROL=/root/.cache/keyring-aZPaEg
GPG_AGENT_INFO=/root/.cache/keyring-aZPaEg/gpg:0:1
GPG_AGENT_INFO=/root/.cache/keyring-aZPaEg/gpg:0:1
GNOME_KEYRING_CONTROL=/root/.cache/keyring-aZPaEg
GPG_AGENT_INFO=/root/.cache/keyring-aZPaEg/gpg:0:1
SSH_AUTH_SOCK=/root/.cache/keyring-aZPaEg/ssh
Connection failure: Connection refused
pa_context_connect() failed: Connection refused

(cinnamon-settings-daemon:4819): power-plugin-WARNING **: Failed set DPMS mode: Display is not DPMS capable
=== xinerama setup Configuration ===
  Clone: false
  Output: Laptop attached to default
     status: on
     width: 1600
     height: 1200
     rate: 0
     primary: true
     position: 0 0
=== auto-configure - xinerama mode Configuration ===
  Clone: false
  Output: Laptop attached to default
     status: on
     width: 1600
     height: 1200
     rate: 0
     primary: true
     position: 0 0
=== Applying Configuration Configuration ===
  Clone: false
  Output: Laptop attached to default
     status: on
     width: 1600
     height: 1200
     rate: 0
     primary: true
     position: 0 0

** (cinnamon-settings-daemon:4819): WARNING **: Failed to get session for pid: GDBus.Error:org.freedesktop.ConsoleKit.Manager.GeneralError: Unable to lookup session information for process '4819'

** (cinnamon-settings-daemon:4819): WARNING **: Failed to get session for pid: GDBus.Error:org.freedesktop.ConsoleKit.Manager.GeneralError: Unable to lookup session information for process '4819'

(cinnamon-settings-daemon:4819): color-plugin-WARNING **: failed to get edid: unable to get EDID for output
[1390927757,000,xklavier.c:xkl_engine_start_listen/]    The backend does not require manual layout management - but it is provided by the application

** (cinnamon-settings-daemon:4819): WARNING **: Failed to connect context: OK

(cinnamon-settings-daemon:4819): color-plugin-WARNING **: unable to get EDID for xrandr-default: unable to get EDID for output

(cinnamon-settings-daemon:4819): color-plugin-WARNING **: failed to reset xrandr-default gamma tables: gamma size is zero

(cinnamon-settings-daemon:4819): color-plugin-WARNING **: no xrandr-ViewSonic Corporation-VA703-4SERIE-QAP070521053 device found: Failed to find output xrandr-ViewSonic Corporation-VA703-4SERIE-QAP070521053

(cinnamon-settings-daemon:4819): color-plugin-WARNING **: no xrandr-ViewSonic Corporation-VX1937 SERIE-S9D104103321 device found: Failed to find output xrandr-ViewSonic Corporation-VX1937 SERIE-S9D104103321

(cinnamon-settings-daemon:4819): color-plugin-WARNING **: unable to get EDID for xrandr-default: unable to get EDID for output

(cinnamon-settings-daemon:4819): color-plugin-WARNING **: failed to reset xrandr-default gamma tables: gamma size is zero
Window manager warning: Software rendering detected: Gallium 0.4 on llvmpipe (LLVM 3.3, 128 bits)

** (polkit-gnome-authentication-agent-1:4901): WARNING **: Unable to register authentication agent: GDBus.Error:org.freedesktop.PolicyKit1.Error.Failed: An authentication agent already exists for the given subject
Cannot register authentication agent: GDBus.Error:org.freedesktop.PolicyKit1.Error.Failed: An authentication agent already exists for the given subject

** (cinnamon-fallback-mount-helper:4889): WARNING **: Failed to get session for pid: GDBus.Error:org.freedesktop.ConsoleKit.Manager.GeneralError: Unable to lookup session information for process '4889'
** Message: applet now removed from the notification area
      JS LOG: About to start Cinnamon
      JS LOG: ACTIVATING SOFTWARE RENDERING
** Message: using fallback from indicator to GtkStatusIcon

** (nm-applet:4892): WARNING **: Failed to register as an agent: (32) Session not found

** (nemo:4890): WARNING **: Can not get _NET_WORKAREA

** (nemo:4890): WARNING **: Can not determine workarea, guessing at layout
      JS LOG: Cinnamon started at Tue Jan 28 2014 16:49:28 GMT+0000 (UTC)
      JS LOG: network applet: Cannot find connection for active (or connection cannot be read)

(cinnamon:4882): Gvc-WARNING **: Failed to connect context: OK
** Message: applet now embedded in the notification area
    JS ERROR: !!!   Exception was: Error: missing path proxifying object
    JS ERROR: !!!     message = '"missing path proxifying object"'
    JS ERROR: !!!     fileName = '"/usr/share/cjs-1.0/dbus.js"'
    JS ERROR: !!!     lineNumber = '321'
    JS ERROR: !!!     stack = '"([object Object],"org.freedesktop.ConsoleKit",null)@/usr/share/cjs-1.0/dbus.js:321
([object Object],"org.freedesktop.ConsoleKit",null)@/usr/share/cjs-1.0/dbus.js:602
([object Object],"org.freedesktop.ConsoleKit",null)@/usr/share/cjs-1.0/dbus.js:599
(null,[object Error])@/usr/share/cinnamon/js/ui/automountManager.js:63
"'
      JS LOG: network applet: Found connection for active

** (gnome-user-share:4975): WARNING **: gnome-user-share cannot be started as root for security reasons.
-- Exception occured in the refresh thread: 'NoneType' object has no attribute 'set_cursor'
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 808, in __bootstrap_inner
    self.run()
  File "/usr/lib/linuxmint/mintUpdate/mintUpdate.py", line 682, in run
    self.wTree.get_widget("window1").window.set_cursor(None)
AttributeError: 'NoneType' object has no attribute 'set_cursor'
```

**Comment by ghost** [comment]
@mbnoimi 
From the log output in between:

```
 ...
 (cinnamon-settings-daemon:4819): power-plugin-WARNING **: Failed set DPMS mode:
 Display is not DPMS capable
 === xinerama setup Configuration ===
   Clone: false
   Output: Laptop attached to default
      status: on
      width: 1600
      height: 1200
 ...
```

Cinnamon seemed to take control of the Desktop "default" resolution at 1600 by 1200. I am unsure where does Cinnamon keeps its screen resolution setting.. But for this kind of issue, it can be solved by changing the screen resolution to desired size, just one time.

Anyway, without relying on application menu, can you try to change the display resolution from right-click > Display properties or something like that?

**Comment by mbnoimi** [comment]
@clearkimura 

> Anyway, without relying on application menu, can you try to change the display resolution from right-click > Display properties or something like that?

As I said above the resolution of Xephyr is too big so I can't do anything with it! There isn't any way to change the resolution without accessing the Control Panel which is unavailable through right click!

Do you've any idea how can fix this issue?

P.S. I consider it as a bug because Xephyr doesn't take the right resolution as I set it for (800x600 by default)

**Comment by ghost** [comment]
@mbnoimi 
This may be a bug, but _only_ affecting distros with Cinnamon desktop environment.

I can assure that this issue _didn't_ happen on other tested environments, including Xfce and KDE, mainly because I have tested myself on these already within several months ago.

And of course, I tested for Cinnamon and this issue did happen.

That time, I had resized the Xephyr window frame (1600 by 1200 resolution) to smaller size than my monitor (1280 by 1024 resolution) until I can see the vertical and horizontal scrollbars. Then, I scroll down to see menu, click control panel, rest is as you know--doing all manually*.

This trick, just do _only_ once, did serve me as workaround, so perhaps you can try this\* (rather than waiting for fix that only affects Cinnamon alone)? 

Edit:  By the way, I have no idea how to solve this at the moment. I don't have anything useful around me, because I am working far away from home. So, no.. this may not be fixed any soon.

**Comment by mbnoimi** [comment]
@clearkimura 

> I had resized the Xephyr window frame (1600 by 1200 resolution) to smaller size than my monitor (1280 by 1024 resolution)

How you did that? 
I tried to resize Xephyr window frame by mouse but it showed fixed (unresizeable)!

> Edit: By the way, I have no idea how to solve this at the moment. I don't have anything useful around me, because I am working far away from home. So, no.. this may not be fixed any soon.

Thanks a lot for interesting.

**Comment by ghost** [comment]
@mbnoimi 
Hi, there. Just to refresh this issue and revisit my experience.

> How you did that?
> I tried to resize Xephyr window frame by mouse but it showed fixed (unresizeable)!

You were absolutely correct. My memory was false. Xephyr window frame is unresizeable as you claimed, whilst I tried to reproduce this issue using old Linux Mint 9 running GNOME 2.x.

To reproduce this issue, we simply run the Xephyr Live session:
1. In Customizer, click on "Desktop" button. A Terminal window will appear.
2. Select whichever session you wish (in your case, you selected "cinnamon")
3. Wait until the Desktop loads. At this point, you probably can only see part of the highest resolution which Cinnamon had automatically selected for your laptop supported resolution i.e. 1600 by 1200 pixels. That being said, my workaround takes place from next step:
4. In Xephyr window, press keyboard combination Ctrl+Shift keys to grab mouse and keyboard. The window frame title will change, and now your mouse and keyboard is only perceived by Xephyr Live session.
5. Press keyboard shortcut Alt+F1 to bring out the main menu (this was my case, GNOME 2.x in Linux Mint 9), of the Xephyr Live session, and _not_ the host system--this was the purpose of previous step. From here, I navigate to "Monitor" or "Display" menu and change the current Desktop resolution to something smaller.

In your case, which I assume newer release of Linux Mint, it is better to use Alt+F2 to bring out Run dialog (or Terminal) and type "gnome-control-center", "cinnamon-control-center" or something similar, depending on which release you are using.

This way, you can reposition the Control Center window to viewable Desktop area then navigate to "Display" setting, rather than depending on the main menu which is found at bottom panel (that hiding outside the viewable area).

To summarize, workaround via GUI exists for this issue as explained above.

**Comment by ghost** [comment]
@mbnoimi 
I am giving up on this issue for several reasons:
- this issue only affects Cinnamon Desktop environment (not sure if recent releases of Cinnamon still having this problem) wherelse other Desktop environment works fine with Xephyr
- we don't know where or how Cinnamon set default Desktop resolution in Xephyr, which auto-selects the highest supported resolution--which hampers any initial effort to fix this issue
- failed to reproduce this issue, using Linux Mint 9 GNOME and Linux Mint 13 Cinnamon (based on Ubuntu 10.04 and 12.04 respectively)--running "Desktop" feature opened Xephyr window at correct Desktop resolution as setting i.e. 800 by 600

Thereby, I am closing this issue until other users can reproduce this issue.
